frontend/app/model/RichEvent.scala

```diff
@@ -81,35 +81,42 @@ object RichEvent {
 
   trait RichEvent {
     val event: EBEvent
-    val imgUrl: String
+    val imgOpt: Option[model.ResponsiveImageGroup]
     val socialImgUrl: String
     val tags: Seq[String]
     val metadata: Metadata
     val imageMetadata: Option[Grid.Metadata]
     val contentOpt: Option[Content]
-    val srcsetOpt: Option[Seq[String]]
-    val availableWidths: String
-    val fallbackImage = views.support.Asset.at("images/event-placeholder.gif")
     val pastImageOpt: Option[Asset]
-
     def deficientGuardianMembersTickets: Boolean
   }
 
   abstract class LiveEvent(image: Option[EventImage], contentOpt: Option[Content]) extends RichEvent {
 
-    private val widths = image.fold(List.empty[Int])(_.assets.map(_.dimensions.width))
-
     val imageMetadata = image.map(_.metadata)
 
-    val imgUrl = image.flatMap(_.assets.headOption).fold(fallbackImage){ asset =>
-      asset.secureUrl.getOrElse(asset.file)
-    }
+    val imgOpt = image.map { imgData =>
+      Some(ResponsiveImageGroup(
+        altText=event.name.toString(),
+        availableImages=imgData.assets.map { asset =>
+          ResponsiveImage(
+            path=asset.secureUrl.getOrElse(asset.file),
+            width=asset.dimensions.width
+          )
+        }
+      ))
+    }.getOrElse(None)
```

Comment:
Had to use a `flatMap` in order to get rid of the `getOrElse(None)`  but that seems to have done the trick.